### ISSUE TITLE
[clang-tidy] "unknown pragma ignored" generated on "pragma pack" in c++20 mode

### ISSUE THREAD

**Issue by vient** [llvm/llvm-project#67380]
Source `test.cpp`
```cpp
#pragma pack(1)
struct S {};
```
Compiling with `clang++-17 -Wall -Werror -std=c++20 -o test.o -c test.cpp` works fine, no errors.

I created `/tmp/test/compile_commands.json` using `-MJ` clang option output:
```json
[
  {
    "directory": "/tmp/test",
    "file": "test.cpp",
    "output": "test.o",
    "arguments": [
      "/usr/lib/llvm-17/bin/clang",
      "-xc++",
      "test.cpp",
      "-o",
      "test.o",
      "--driver-mode=g++",
      "-Wall",
      "-Werror",
      "-std=c++20",
      "-c",
      "--target=x86_64-pc-linux-gnu"
    ]
  }
]
```
Running `clang-tidy-17 test.cpp` in /tmp/test folder produces following output
```console
$ clang-tidy-17 test.cpp
1 error generated.
Error while processing /tmp/test/test.cpp.
test.cpp:1:9: error: unknown pragma ignored [clang-diagnostic-unknown-pragmas]
    1 | #pragma pack(1)
      |         ^
Found compiler error(s).
```
Error is only produced in `c++20` and newer modes, like `c++2b`, but not in `c++17` or older.

I am using latest 17 build from LLVM apt repo
```console
$ clang++-17 --version
Ubuntu clang version 17.0.2 (++20230925113614+481358974fb0-1~exp1~20230925113629.44)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

$ clang-tidy-17 --version
Ubuntu LLVM version 17.0.2
  Optimized build.
```

**Comment by vient** [comment]
Can't get this error on CE https://godbolt.org/z/4vPYY6aY4, does it mean that this is not present in trunk?

**Comment by PiotrZSL** [comment]
I couldn't reproduce it with clang-tidy 17 or trunk on ubuntu.

**Comment by PiotrZSL** [comment]
Got it with:
```
clang-tidy-17 test.cpp --extra-arg=-std=c++20
1 warning generated.
/fpwork/llvm-project/test.cpp:1:9: warning: unknown pragma ignored [clang-diagnostic-unknown-pragmas]
    1 | #pragma pack(1)
      |         ^
```
or
```
clang-tidy-17 test.cpp -- -std=c++20 -Wall
1 warning generated.
/fpwork/llvm-project/test.cpp:1:9: warning: unknown pragma ignored [clang-diagnostic-unknown-pragmas]
    1 | #pragma pack(1)
      |         ^
```

With trunk version is reproduced only when '--enable-module-headers-parsing' is used.
This is known issue caused by two preprocessors created for C++20 and above.
This functionality is disabled by default in clang-tidy 18.

**Comment by vient** [comment]
Ok, thanks for info, added this to my `.clang-tidy`
```yaml
ExtraArgs:
- -Wno-unknown-pragmas  # TODO: remove this after switch to clang-tidy-18
```

**Comment by spt29** [comment]
Is this the fix? 992fa7be3438bce61cfba7c0e0b07d9d1ddcc05b Any chance of a backport into the 17.x release branch?
